pkg/block/block.go

```diff
@@ -126,6 +128,32 @@ func cleanUp(logger log.Logger, bkt objstore.Bucket, id ulid.ULID, err error) er
 	return err
 }
 
+// MarkForDeletion creates a file
+// which stores information about when the block was marked for deletion.
+func MarkForDeletion(ctx context.Context, logger log.Logger, bkt objstore.Bucket, id ulid.ULID) error {
+	compactorMetaExists, err := objstore.Exists(ctx, bkt, path.Join(id.String(), metadata.CompactorMetaFilename))
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("check compactor meta for id %s in bucket", id.String()))
+	}
+	if compactorMetaExists {
+		level.Info(logger).Log("msg", "compactor-meta already exists for block", "id", id.String())
+		return nil
+	}
+
+	compactorMeta, err := json.Marshal(metadata.CompactorMeta{
+		ID:           id,
+		DeletionTime: time.Now().Unix(),
+	})
+	if err != nil {
+		return errors.Wrap(err, "compactor meta json marshalling")
+	}
+
+	if err := bkt.Upload(ctx, path.Join(id.String(), metadata.CompactorMetaFilename), bytes.NewReader(compactorMeta)); err != nil {
+		return errors.Wrap(err, "upload meta file to debug dir")
+	}
+	return nil
```

Comment:
Elsewhere you wrote:

> I am not deleting file from DebugCompactorMetas in bucket because that can be used to gain information about when the block was marked for deletion.

If you want to preserve that information somewhere, I would put a log message here saying that "block <ID> has been marked for deletion". That way it will be available in the log, and have timestamp as well.

Reply:
Yes :+1: 
Also we need metric